lightningd/bitcoind.c

```diff
@@ -997,6 +997,103 @@ void bitcoind_getclientversion(struct bitcoind *bitcoind)
 			  "getnetworkinfo", NULL);
 }
 
+/* Mutual recursion */
+static bool process_getblockchaininfo(struct bitcoin_cli *bcli);
+
+static void retry_getblockchaininfo(struct bitcoind *bitcoind)
+{
+	assert(!bitcoind->synced);
+	start_bitcoin_cli(bitcoind, NULL,
+			  process_getblockchaininfo,
+			  false, BITCOIND_LOW_PRIO, NULL, NULL,
+			  "getblockchaininfo", NULL);
+}
+
+/* Given JSON object from getblockchaininfo, are we synced?  Poll if not. */
+static void is_bitcoind_synced_yet(struct bitcoind *bitcoind,
+				   const char *output, size_t output_len,
+				   const jsmntok_t *obj,
+				   bool initial)
+{
+	const jsmntok_t *t;
+	unsigned int headers, blocks;
+	bool ibd;
+
+	t = json_get_member(output, obj, "headers");
+	if (!t || !json_to_number(output, t, &headers))
+		fatal("Invalid 'headers' field in getblockchaininfo '%.*s'",
+		      (int)output_len, output);
+
+	t = json_get_member(output, obj, "blocks");
+	if (!t || !json_to_number(output, t, &blocks))
+		fatal("Invalid 'blocks' field in getblockchaininfo '%.*s'",
+		      (int)output_len, output);
+
+	t = json_get_member(output, obj, "initialblockdownload");
+	if (!t || !json_to_bool(output, t, &ibd))
+		fatal("Invalid 'initialblockdownload' field in getblockchaininfo '%.*s'",
+		      (int)output_len, output);
+
+	if (ibd) {
+		if (initial)
+			log_unusual(bitcoind->log,
+				    "Waiting for initial block download"
+				    " (this can take a while!)");
```

Comment:
Won't really work since by that time we already have processed the current blockchain head - 14, meaning we have inserted a high-water-mark, and we'll restart scanning from that height. We might need a better heuristic for "not-yet-locked-in" start height allowing us to reset that high-water-mark on restart. But that's for another PR I'd say.